src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs

```diff
@@ -20,27 +21,28 @@ public static partial class ApiClientExtensions
     public static IServiceCollection AddCoinGeckoClient(
         this IServiceCollection services, IConfiguration configuration)
     {
-        var section = configuration.GetSection(nameof(CoinGeckoApiConfiguration));
-        services.Configure<CoinGeckoApiConfiguration>(section);
-
-        var typedConfig = section.Get<CoinGeckoApiConfiguration>()!;
-
-        return services.AddCoinGeckoClient(typedConfig);
+        var apiConfiguration = configuration.GetConfiguration<CoinGeckoApiConfiguration>();
+        var cacheConfiguration = configuration.GetConfiguration<RedisCacheConfiguration>();
+        return services.AddCoinGeckoClient(apiConfiguration, cacheConfiguration);
     }
 
     public static IServiceCollection AddCoinGeckoClient(
-        this IServiceCollection services, CoinGeckoApiConfiguration apiConfiguration)
+        this IServiceCollection services,
+        CoinGeckoApiConfiguration apiConfiguration,
+        RedisCacheConfiguration cacheConfiguration)
     {
+        // api configuration
         services.AddSingleton(apiConfiguration);
-        services.AddSingleton(services => new ClientConfigurator(apiConfiguration));
-
-        services.AddSingleton<ICoinGeckoClient, CoinGeckoClient>();
-        services.AddTransient<CachedHttpClientHandler>();
+        services.AddSingleton(new ClientConfigurator(apiConfiguration));
 
+        // required common services
         services.AddSingleton<IDateTimeProvider, DateTimeProvider>();
-        services.AddSingleton<ISemaphore>(new Semaphore(new SemaphoreSlim(1, 1)));
+        services.AddDistributedCache(cacheConfiguration);
         services.AddMemoryCache();
```

Comment:
I'm not overly fond of setting up these "external" services here, but I understand how it's easier to consume this api client.